tsfc/coffee_mode.py

```diff
@@ -45,15 +80,21 @@ def predicate(index):
     expressions = replace_delta(expressions)
     expressions = remove_componenttensors(expressions)
     expressions = replace_division(expressions)
-    return optimise_expressions(expressions, argument_multiindices)
+    argument_indices = tuple(itertools.chain(*argument_multiindices))
+
+    def conditional_predicate(argument_indices, node):
+        return len(set(node.free_indices) & argument_indices) >= 1
```

Comment:
If you can rewrite this predicate in terms of the classifier, then the `handle_conditional` call below may go to `collect_monomials`.